File: appointment/lib/controllers/auth_controller.dart
// lib/controllers/auth_controller.dart

import 'package:get/get.dart';

import '../models/user_model.dart';
import '../service/auth_service.dart';

class AuthController extends GetxController {
  final RxBool isLoading = false.obs;
  final Rxn<UserModel> currentUser = Rxn<UserModel>();
  String? authToken;

  bool get isLoggedIn => authToken != null && currentUser.value != null;

  Future<void> login({
    required String mobile,
    required String password,
  }) async {
    if (mobile.isEmpty || password.isEmpty) {
      Get.snackbar('Error', 'Mobile and password are required');
      return;
    }

    try {
      isLoading.value = true;

      final loginResponse = await AuthService.login(
        mobile: mobile,
        password: password,
      );

      authToken = loginResponse.token;
      currentUser.value = loginResponse.user;

      Get.snackbar('Success', 'Login successful');

      // Go to home
      Get.offAllNamed('/home');
    } catch (e) {
      Get.snackbar('Login Failed', e.toString());
    } finally {
      isLoading.value = false;
    }
  }

  // NEW: register
  Future<void> register({
    required String name,
    required String mobile,
    required String password,
    String? email,
  }) async {
    if (name.isEmpty || mobile.isEmpty || password.isEmpty) {
      Get.snackbar('Error', 'Name, mobile and password are required');
      return;
    }

    try {
      isLoading.value = true;

      final res = await AuthService.register(
        name: name,
        mobile: mobile,
        password: password,
        email: email,
      );

      Get.snackbar('Success', res.message.isNotEmpty
          ? res.message
          : 'Registered successfully');

      // After successful register, go back to Login screen
      Get.offAllNamed('/login');
    } catch (e) {
      Get.snackbar('Register Failed', e.toString());
    } finally {
      isLoading.value = false;
    }
  }
}

File: appointment_booking/api/src/controllers/authController.js
const User = require("../models/User");


exports.register = async (req, res) => {
  try {
    const { name, mobile, password, email } = req.body;

    if (!name || !mobile || !password) {
      return res
        .status(400)
        .json({ message: "name, mobile and password are required" });
    }

    const existing = await User.findOne({ mobile });
    if (existing) {
      return res.status(409).json({ message: "Mobile number already registered" });
    }

    // NOTE: in real life, hash the password before saving
    const user = await User.create({ name, mobile, password, email });

    return res.status(201).json({
      message: "User registered successfully",
      user: {
        _id: user._id,
        name: user.name,
        mobile: user.mobile,
        email: user.email,
      },
    });
  } catch (err) {
    console.error("Error in register:", err);
    return res.status(500).json({ message: "Server error" });
  }
};


exports.login = async (req, res) => {
  try {
    const { mobile, password } = req.body;

    if (!mobile || !password) {
      return res
        .status(400)
        .json({ message: "mobile and password are required" });
    }

    const user = await User.findOne({ mobile });

    if (!user) {
      return res.status(401).json({ message: "Invalid mobile or password" });
    }

    if (user.password !== password) {
      return res.status(401).json({ message: "Invalid mobile or password" });
    }

    const dummyToken = `dummy-auth-token-${user._id}`;

    return res.json({
      message: "Login successful",
      token: dummyToken,
      user: {
        _id: user._id,
        name: user.name,
        mobile: user.mobile,
        email: user.email,
      },
    });
  } catch (err) {
    console.error("Error in login:", err);
    return res.status(500).json({ message: "Server error" });
  }
};

